### ISSUE TITLE
ambiguous unhinted method-as-function should warn about adding metadata

### ISSUE THREAD

**Issue by borkdude** [clj-kondo/clj-kondo#2281]
When the metadata here is missing:

```
(apply ^[] String/toUpperCase ["hi"])
```

clj-kondo could show the compiler error message (and possibly even the desired metadata)

**Comment by borkdude** [comment]
To solve this issue we need to add the arities to the cache. Currently we only have flags.

**Comment by borkdude** [comment]
Not sure what the original problem was, this works though: `(apply ^[] String/.toUpperCase ["hi"])`
